### PR TITLE
Use Nixpkgs version of FiPy expression

### PR DESCRIPTION
**Pull request by wd15**
shell.nix replaced default.nix to create a development version of FiPy
based on the Nixpkgs FiPy expression

### CODE REVIEW

**Comment by wd15** [comment]
This supersedes #658

File: documentation/NIX-README.rst [hunk 1]
```diff
@@ -48,8 +45,9 @@ Additional Packages
 -------------------
 
 To install additional packages available from Nixpkgs_ include them in
-the `buildInputs` list in `nix/build.nix` (in the `nix`
-subdirectory of the base :term:`FiPy` directory).
+the `nativeBuildInputs` list in `shell.nix` (in the `nix` subdirectory
```
Comment (tkphd) [inline documentation/NIX-README.rst]: Directory does not exist.

Comment (wd15) [reply documentation/NIX-README.rst]: Done

**Comment by wd15** [comment]
@tkphd, do you get that same error with a Conda build of FiPy? If not, we need to try and debug.

**Comment by wd15** [comment]
@tkphd, you're probably using the pyamg solvers. Can you try rerunning the tests with `python setup.py test --scipy`?

**Comment by tkphd** [comment]
@wd15 right you are! Please update the docs with that solver specified for Nix.

**Comment by wd15** [comment]
@tkphd, I'm not sure that's a good idea. It sort of mixes issues up. The pyamg versus scipy solver order is an issue with FiPy in Python 3 not necessarily with Nix. Also the failure in the test with pyamg is probably not only with the Nix build as well.

**Comment by tkphd** [comment]
True, OK.
Is the "conda3_env" build failure on CircleCI also out-of-scope?

**Comment by wd15** [comment]
Must be. The only tests that this change can impact would be a documentation build.

**Comment by guyer** [comment]
Something broke in the conda stack three days ago, causing conda3_env to fail, between https://circleci.com/gh/usnistgov/fipy/759 and https://circleci.com/gh/usnistgov/fipy/773. Needs to be figured out and fixed, but has nothing to do with this pull request.

**Comment by wd15** [comment]
That's bonkers. Can you do anything about that by pinning things or is that out of your control?

> Something broke in the conda stack three days ago, causing conda3_env to fail, between https://circleci.com/gh/usnistgov/fipy/759 and https://circleci.com/gh/usnistgov/fipy/773. Needs to be figured out and fixed, but has nothing to do with this pull request.

**Comment by wd15** [comment]
@guyer, this should work no on Darwin. Please give it a whirl.

**Comment by wd15** [comment]
I would suggest rebasing this before merging

**Pull request merged**